src/ir/iteration.h

```diff
@@ -57,8 +70,11 @@ template<class Specific> class AbstractChildIterator {
 
     void operator++() { index++; }
 
-    Expression*& operator*() {
-      return *parent.children[parent.mapIndex(index)];
+    // Depending on which specilization of DereferenceKind we use, this returns
+    // either Expression* or Expression**.
```

Comment:
Why do we need both `Expression*&` and `Expression**` return types? They seem fairly similar.

```suggestion
    // either Expression*& or Expression**.
```